consensus/src/dag/tests/integration_tests.rs

```diff
@@ -0,0 +1,232 @@
+use super::dag_test;
+use crate::{
+    dag::{bootstrap::bootstrap_dag, CertifiedNode},
+    network::{DAGNetworkSenderImpl, IncomingDAGRequest, NetworkSender},
+    network_interface::{ConsensusMsg, ConsensusNetworkClient, DIRECT_SEND, RPC},
+    network_tests::{NetworkPlayground, TwinId},
+    test_utils::{consensus_runtime, MockPayloadManager, MockStorage},
+    util::time_service::{ClockTimeService, TimeService},
+};
+use aptos_channels::{aptos_channel, message_queues::QueueStyle};
+use aptos_config::network_id::{NetworkId, PeerNetworkId};
+use aptos_consensus_types::common::Author;
+use aptos_logger::debug;
+use aptos_network::{
+    application::interface::NetworkClient,
+    peer_manager::{conn_notifs_channel, ConnectionRequestSender, PeerManagerRequestSender},
+    protocols::{network::{self, Event, NetworkEvents, NewNetworkEvents, NewNetworkSender}, wire::handshake::v1::ProtocolIdSet}, transport::ConnectionMetadata, ProtocolId,
+};
+use aptos_types::{
+    epoch_state::EpochState,
+    validator_signer::ValidatorSigner,
+    validator_verifier::{random_validator_verifier, ValidatorVerifier},
+};
+use claims::assert_gt;
+use futures::{stream::{AbortHandle, select, Select}, StreamExt};
+use futures_channel::mpsc::UnboundedReceiver;
+use maplit::hashmap;
+use std::sync::Arc;
+use tokio::runtime::Runtime;
+
+struct DagBootstrapUnit {
+    nh_abort_handle: AbortHandle,
+    df_abort_handle: AbortHandle,
+    dag_rpc_tx: aptos_channel::Sender<Author, IncomingDAGRequest>,
+    network_events: Box<Select<NetworkEvents<ConsensusMsg>, aptos_channels::Receiver<Event<ConsensusMsg>>>>,
+}
+
+impl DagBootstrapUnit {
+    fn new(
```

Comment:
nit: when I read new, I typically expect the return is a single Self, probably should have a different name (I was confused when looking at the unzip below)